src/commands/dev/socket.rs

```diff
@@ -42,26 +52,30 @@ pub async fn listen(session_id: &str) -> Result<(), failure::Error> {
             }
         })
     };
-    pin_mut!(keep_alive_to_ws, print_ws_messages);
-    future::select(keep_alive_to_ws, print_ws_messages).await;
-    Ok(())
+
+    // run the heartbeat and message printer in parallel
+    tokio::select! {
+        _ = keep_alive_to_ws => { Ok(()) }
+        _ = print_ws_messages => { Ok(()) }
+    }
```

Comment:
this is cleaner than using future::select and pin_mut

Reply:
note that the utility of `tokio::select` is to run until one of its blocks returns and then abandon the other. not certain that's the precise intention here, perhaps  `try_join` would be more in  line  with intent?

Reply:
nah - select is what we want here, printing ws messages will be the first one to go because keep_alive never returns. want them to just run indefinitely

Reply:
hrm. k.